api/models/Community.js

```diff
@@ -164,5 +165,19 @@ module.exports = bookshelf.Model.extend({
       var slackMessage = Slack.textForNewPost(post, community)
       return Slack.send(slackMessage, community.get('slack_hook_url'))
     })
+  },
+
+  getNewBetaAccessCode: function () {
+    const test = code => Community.where({beta_access_code: code}).fetch()
+    const loop = code =>
+      test(code)
+      .then(community => {
+        if (community) {
+          return loop(randomstring.generate({length: 6, charset: 'alphanumeric'}))
+        } else {
+          return code
+        }
+      })
+    return loop(randomstring.generate({length: 6, charset: 'alphanumeric'}))
```

Comment:
you could avoid repeating the `generate` line by just doing it in the beginning of `loop`